pennylane/interfaces/tensorflow.py

```diff
@@ -100,25 +95,25 @@ def _to_tensors(x):
     return tf.convert_to_tensor(x)
 
 
-def _res_restructured(res, tapes, shots=None):
+def _res_restructured(res, tapes, legacy_shots: Shots):
     """
     Reconstruct the nested tuple structure of the output of a list of tapes
     """
-    shot_vector = isinstance(shots, Sequence)
-    num_copies = _get_num_copies(shots) if shot_vector else 1
-
     start = 0
     res_nested = []
     for tape in tapes:
+        tape_shots = legacy_shots or tape.shots or Shots(1)

```

Comment:
that is indeed the reason. in analytic mode, this should use `1` for grouping purposes below. in fact, I added `__bool__` to `Shots` so that even if `tape.shots` is `Shots(None)`, this evaluates to `Shots(1)`. This is to preserve previous behaviour